sound/soc/sof/sof-acpi-dev.c

```diff
@@ -149,13 +149,8 @@ static void sof_acpi_fw_cb(const struct firmware *fw, void *context)
 	}
 
 	/* register PCM and DAI driver */
-	priv->pdev_pcm =
-		platform_device_register_data(dev, "sof-audio", -1,
-					      sof_pdata, sizeof(*sof_pdata));
-	if (IS_ERR(priv->pdev_pcm)) {
-		dev_err(dev, "Cannot register device sof-audio. Error %d\n",
-			(int)PTR_ERR(priv->pdev_pcm));
-	}
+	sof_create_platform_device(priv);
```

Comment:
I suggest your only add the function sof_create_platform_device in utils.c and leave the callbacks for ACPI/SPI/PCI alone since you remove them in the next patch.

Reply:
@keyonjie can you improve this, this creates changes in 3 files for no good reason. Just add the new routine in utils and leave acpi/pci/spi alone